layouts/_default/_markup/render-link.html

```diff
@@ -1,11 +1,14 @@
 {{ $isRedirected := findRE "^(|../|/docs/(languages|platforms)/.+?/)(api|daily-builds)(/.*)?(\\?.*)?$" .Destination -}}
 {{ $isExternal := or (hasPrefix .Destination "http") $isRedirected -}}
+{{/* Until Hugo supports hook params (https://github.com/gohugoio/hugo/issues/6670), we'll inspect .Text. */ -}}
+{{ $noExternalIcon := in .Text "hk-no-external-icon" -}}
```

Comment:
Of course, we could inspect `.Title`, but the use of `.Text` fits our current use case (as will become evident in the followup PR).